### ISSUE TITLE
[SF Migration] Rework access denied error messages

### ISSUE THREAD

**Issue by matks** [PrestaShop/PrestaShop#12235]
Currently, we have the following error messages usecases for migrated pages:

- when someone attempts to display a page forbidden to him: "Access denied." or "You do not have permission to view this."
- when someone attempts to create a new item but he was not granted the right to create: "You do not have permission to add this." or "You do not have permission to create this."
- when someone attempts to modify an item but he was not granted the right to edit: "You do not have permission to edit this."
- when someone attempts to delete an item but he was not granted the right to delete : "You do not have permission to delete this." 
- when someone attempts to modify some shop settings through a form but he was not granted the right to modify* : "You do not have permission to edit this." or "You do not have permission to update this."

*For a settings form, you must have either the "create", "update" or "delete" rights to use it.

**Comment by matks** [comment]
We are discussing the possibility of adding a new message "You need permission to perform this action."

**Comment by LouiseBonnard** [comment]
w/ @TristanLDD

**Comment by LouiseBonnard** [comment]
Actually, the idea is a bit broader, but I don't know if this can be done at this point. If we want the software to be consistent, it would be nice to have no negative turns of phrase but positive ones instead. Like _You need permission to perform this action_ instead of _You do not have permission to perform this action_.

**Comment by matks** [comment]
@LouiseBonnard A wording is a wording to us, it does not require more time to use `You need permission to perform this action` .

I think we can even provide which permission is required. Example: "You need CREATE or READ permission to perform this action"

**Comment by matks** [comment]
Do you think you can make a clear rule for us to follow such as

| Triggering condition | Error message |
----- | -----
| When someone attempts to display a page forbidden to him | "Access denied." |
| When someone attempts to create a new item but he was not granted the right to create | ... |
| When someone attempts to modify an item but he was not granted the right to edit | ... |
| ... | ... |

**Comment by LouiseBonnard** [comment]
Absolutely! But in case we decide to turn the negative form into a positive one, I guess that all these kind of occurrence will have to be replaced by positive forms. I can fill this table with better wording suggestions corresponding to each triggering condition. ;-)

**Comment by LouiseBonnard** [comment]
So, here is what I suggest : 

| Triggering condition | Error message - to localize in Admin.Notifications.Error |
----- | -----
| When s.o. attempts to display a page forbidden to him | cf. #10780? |
| When s.o. attempts to create a new item but he was not granted the right to create | You need permission to create this. |
| When s.o. attempts to modify an item but he was not granted the right to edit | You need permission to edit this. |
| When s.o. attempts to delete an item but he was not granted the right to delete | You need permission to delete this. |
| When s.o. attempts to modify some shop settings through a form but he was not granted the right to modify | You need permission to modify/update this. |

One point though: we are not detailing the action nor the next steps at this point, it allows us to have a wording reusable - it works just fine without going further. @matks, @TristanLDD, is it alright for you both?

**Comment by TristanLDD** [comment]
Yep it's ok for me ! :)
